numpy/core/_internal.py

```diff
@@ -631,4 +631,17 @@ class TooHardError(RuntimeError):
     pass
 
 class AxisError(ValueError, IndexError):
-    pass
+    """ Axis supplied was invalid. """
```

Comment:
Now with a docstring, which Ipython shows when you type an exception name. This is in a similar style to the builtin ones.